core/src/main/java/io/smallrye/openapi/api/annotations/SmallRyeOpenApiPath.java

```diff
@@ -0,0 +1,23 @@
+package io.smallrye.openapi.api.annotations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to include a class to the path section of the generated OpenAPI specification, with the specified value as the path.
+ * <p>
+ * Currently, this annotation is only supported by the JAX-RS extension.
+ * </p>
+ * 
+ * @author Anar Sultanov (anar@sultanov.dev)
+ */
+@Target({ ElementType.TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface SmallRyeOpenApiPath {
```

Comment:
BTW did you look into: https://quarkus.io/guides/resteasy#includeexclude-jax-rs-classes-with-build-time-conditions?

Maybe these could be combined to generate the docs during build time, so the profile enabled for build, but disabled for prod?

Reply:
Do you mean a combination of If/Unless annotations? Does that mean that once it's excluded at build time, there is a way to include it at runtime again? Currently, the OpenAPI scanner doesn't know about those annotations so it includes them anyway [1], but that may not be the case forever :-)

[1] https://github.com/quarkusio/quarkus/issues/16218

Reply:
Rather than introducing an new annotation, I would suggest allowing some config, that can define classes that needs to be scanned as paths. So the config will contain the path in the config key and the value will be the class. w.d.y.t ?

Reply:
> Do you mean a combination of If/Unless annotations? Does that mean that once it's excluded at build time, there is a way to include it at runtime again? Currently, the OpenAPI scanner doesn't know about those annotations so it includes them anyway [1], but that may not be the case forever :-)
> 
> [1] [quarkusio/quarkus#16218](https://github.com/quarkusio/quarkus/issues/16218)

Yes, I suspected that OpenAPI might not be looking into it. The classes are still there for runtime; they are just not configured with RESTEasy.

Reply:
There is an implementation present, but we don't want RESTEasy to manage it. I do like @phillip-kruger's suggestion as well and we should explore that one.

Reply:
I also like this suggestion because then you don't have to change the code. I will try to implement this configuration option and update the PR.

Reply:
A small change to the suggestion is to put the fully-qualified class name in the property key and have the value be the path. I think this may be necessary since multiple classes could have the same `@Path` and this would mirror that functionality.

Reply:
@MikeEdgar @phillip-kruger I tried to figure out how to implement this option using the existing ones as an example, but unfortunately there are no other map parameters, and what is there confuses me a little.

For example, there is the following parameter in `GenerateSchemaMojo`:
```
    /**
     * Prefix of the configuration property to specify an alternative list of servers to service all operations in a path
     */
    @Parameter(property = "pathServers")
    private List<String> pathServers;
```
Which is passed in the properties map to `MavenConfig`:
```
    private OpenApiDocument generateSchema(IndexView index) throws IOException, DependencyResolutionRequiredException {
        // ...
        OpenApiConfig openApiConfig = new MavenConfig(getProperties());
        // ...
    }
    
    private Map<String, String> getProperties() throws IOException {
        Map<String, String> cp = new HashMap<>();
        // ...
        addToPropertyMap(cp, OASConfig.SERVERS_PATH_PREFIX, pathServers);
        // ...
        return cp;
    }
 ```
 What confuses me is that the list of the path servers is joined into a string and added to the map with the given key:
 ```
     private void addToPropertyMap(Map<String, String> map, String key, List<String> values) {
        if (values != null && !values.isEmpty()) {
            map.put(key, String.join(",", values));
        }
    }
```
But when they are read inside `MavenConfig`, the key is concatenated with the path:
```
    public List<String> pathServers(String path) {
        return asCsvList(properties.getOrDefault(OASConfig.SERVERS_PATH_PREFIX + path, null));
    }
```
So what am I missing and where does this path come from? The same goes for the `operationServers` parameter.

Reply:
@anarsultanov , the closest thing might be the `getSchemas` method in `OpenApiConfigImpl`.

Reply:
This implementation gets config values from `org.eclipse.microprofile.config.Config` and it is not the same implementation that is used in the maven plugin (`MavenConfig`) which I mentioned above and which in my opinion, doesn't quite work correctly (at least the parameters I mentioned) unless I'm missing something.

Reply:
I've managed to implement configuration property and make library to process specified classes (https://github.com/smallrye/smallrye-open-api/pull/1369)

The only thing I'm struggling with is using the path specified in the config. I tried updating the `JaxRsParameterProcessor` with something like this:
```
    @Override
    protected void processPathParameters(ClassInfo resourceClass, MethodInfo resourceMethod, ResourceParameters parameters) {
        ClassInfo resourceMethodClass = resourceMethod.declaringClass();

        // This is what I added
        readConfigurationPathParameter(resourceClass);
        
        /*
         * Phase I - Read class fields, constructors, "setter" methods not annotated with JAX-RS
         * HTTP method. Check both the class declaring the method as well as the resource
         * class, if different.
         */
        readParametersInherited(resourceMethodClass, null, false);

        if (!resourceClass.equals(resourceMethodClass)) {
            /*
             * The resource class may be a subclass/implementor of the resource method class. Scanning
             * the resource class after the method's class allows for parameter details to be overridden
             * by annotations in the subclass.
             */
            readParameters(resourceClass, null, true);
        }

        parameters.setPathItemParameters(getParameters(resourceMethod));
    }

    private void readConfigurationPathParameter(ClassInfo resourceClass) {
        FrameworkParameter frameworkParam = JaxRsParameter.PATH_PARAM.parameter;
        String path = scannerContext.getConfig().getScanResourceClasses()
                .get(OpenApiConstants.SCAN_RESOURCE_CLASS_PREFIX + resourceClass.name().toString());
        if (path != null) {
            readParameter(
                    new ParameterContextKey(path, frameworkParam.location, frameworkParam.defaultStyle),
                    null,
                    frameworkParam,
                    path,
                    resourceClass,
                    false);
        }
    }
```
But unfortunately it doesn't work.

Reply:
@anarsultanov , let's close this one and continue the discussion on #1369